R/pool.R

```diff
@@ -45,118 +53,178 @@ Pool <- R6Class("Pool",
       }
       reg.finalizer(self,
         function(self) {
-          protectDefaultScheduler({
-            freeEnv <- private$freeObjects
-            if (length(freeEnv) > 0) {
-              self$close()
-            }
-          })
+          if (self$valid) {
+            warning("Closing leaked pool.")
```

Comment:
Yeah, we can't just drop the warning/message, I think... Good point about finalization order. Well, so far, I haven't gotten in trouble, so whatever it's doing (I'm assuming a mix of both) seems to end up in a clean result when all is said and done. I'm really not all that happy with just relying on "whatever it's doing," but is there any other way we could do this?